src/models/user.js

```diff
@@ -0,0 +1,69 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+const schema = mongoose.Schema;
+const saltRounds = 10;
+
+const userSchema = new schema({
+  name: {
+    type: String,
+    required: true,
+    minLength: 3,
+    maxLength: 20,
+  },
+  birthdate: {
+    type: Date,
+    required: true,
+  },
+  email: {
+    type: String,
+    required: true,
+    validate: {
+      validator: (value) => {
+        // Regular expression for email validation
+        const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+        return emailRegex.test(value);
+      },
+      message: (props) => `${props.value} is not a valid email address!`,
+    },
+  },
+  age: {
+    type: Number,
+    required: true,
+    min: 0,
+    max: 100,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+});
+// To Encrypt Password while adding new data
+userSchema.pre("save", async function (value) {
```

Comment:
use arrow function syntax

Reply:
Arrow function not work due to this is access globle data so it's not working hear